chain-extensions/unified-accounts/src/lib.rs

```diff
@@ -54,17 +57,15 @@ where
         })? {
             GetEvmAddress => {
                 let account_id: T::AccountId = env.read_as()?;
-
-                let base_weight = <T as frame_system::Config>::DbWeight::get().reads(1);
-                env.charge_weight(base_weight)?;
+                // charge weight
+                env.charge_weight(UAWeight::<T>::uam_to_h160())?;
                 // write to buffer
                 UA::to_h160(&account_id).using_encoded(|r| env.write(r, false, None))?;
             }
             GetEvmAddressOrDefault => {
                 let account_id: T::AccountId = env.read_as()?;
-
-                let base_weight = <T as frame_system::Config>::DbWeight::get().reads(1);
-                env.charge_weight(base_weight)?;
+                // charge weight
+                env.charge_weight(UAWeight::<T>::uam_to_h160_or_default())?;
```

Comment:
One part I dislike here is that you assume that internal implementation of `UA` is the same as yours in this trait.

`UAWeight::<T>::uam_to_h160_or_default())` refers to the weight of `UnifiedAccountMapper` being used. But the implementation below this is _custom_.

Correct way to handle this would be for the mapper trait to ALSO return enum as part of the result.

Reply:
I think for custom implementation it's better if we use hardcoded weights (1 db read) since we know the exactly know implementation without any assumption.

All the use of `xxx_or_default()` trait methods only needs an address and making it return an enum will make it unnecessarily verbose to use it.


```rust
// 100% use case of UA::xx_ot_default() method
let address = UA::to_h160_or_default(&account_id)
```

```rust
// unnecessarily verbose
let address = match UA::to_h160_or_default(&account_id) {
     Mapped(a) => a
     Default(a) => a
}
```

Reply:
I'm not sure I understand how can this be better.

You're relying on benchmarks of _something_, but have different code below.
The _something_ implementation can be changed without any requirements to change the code in the chain extension.
That's wrong.

For the code example you provided, you can implement `From/Into` trait on the result enum so user doesn't have to use `match` syntax.

Reply:
Hmm, `into()` will work. I'll change it.